### PR TITLE
fix cuda compile error with nse_net

### PR DESCRIPTION
**Pull request by zhichen3**

### CODE REVIEW

**Comment by zingale** [comment]
can you tell me the command you are using to build this?

**Comment by zingale** [comment]
one issue is that I don't think any of the hybrj stuff is marked with AMREX_GPU_HOST_DEVICE AMREX_INLINE

**Comment by zhichen3** [comment]
I think it's working. Whenever we use `hybrj`, we just need to create two functions named `jcn` and `fcn` in the header file. 
I can't test it right now since perlmutter is down.

**Comment by zhichen3** [comment]
tested on groot, seems to work.

File: util/hybrj/hybrj.H [hunk 1]
```diff
@@ -12,26 +12,34 @@
...
-           const T& data,
-           std::function<void(amrex::Array1D<Real, 1, neqs>&,
-                              amrex::Array1D<Real, 1, neqs>&,
-                              const T&, int&)> fcn,
-           std::function<void(amrex::Array1D<Real, 1, neqs>&,
-                              amrex::Array2D<Real, 1, neqs, 1, neqs>&,
-                              const T&, int&)> jcn) {
+           const T& data
+	   //,
+           // std::function<void(amrex::Array1D<Real, 1, neqs>&,
```
Comment (zingale) [inline util/hybrj/hybrj.H:19]: can you just remove these instead of commenting them out?

File: util/hybrj/hybrj.H [hunk 2]
```diff
@@ -12,26 +12,34 @@
...
+           //                    amrex::Array2D<Real, 1, neqs, 1, neqs>&,
+           //                    const T&, int&)> jcn
+	   ) {
 
     // the purpose of hybrj is to find a zero of a system of
     // n nonlinear functions in n variables by a modification
     // of the powell hybrid method. the user must provide a
     // subroutine which calculates the functions and the jacobian.
     //
+    // Supposely we ask user to supply fcn and jcn, where
```
Comment (zingale) [inline util/hybrj/hybrj.H:32]: change this to say that we require fcn and jcn to be fined in the program unit where we include this header

File: util/hybrj/hybrj.H [hunk 3]
```diff
@@ -12,26 +12,34 @@
...
     // fcn jcn are the names of the user-supplied subroutine which
     // calculates the functions and the jacobian.
     //
     // void fcn(Array1D<Real, 1, neqs>& x, Array1D<Real, 1, neqs>& fvec, int& iflag)
     // void jcn(Array1D<Real, 1, neqs>& x, Array2D<Real, 1, neqs, 1, neqs> fjac, int& iflag)
     //
+    // However, right now we run into issues when compiling GPU.
+    // So right now we use global function fcn and jcn to calculate
+    // the function and the jacobian.
+    //
```
Comment (zingale) [inline util/hybrj/hybrj.H:42]: all of this can be removed (starting with line 33)

**Review by zingale — APPROVED** [review]

**Pull request merged**